spec/javascripts/Navigation/header/SessionMenu.spec.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import Enzyme, { shallow } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+import { SessionMenu } from 'Navigation/components/header/SessionMenu'
+import { Avatar } from '@patternfly/react-core'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+const props = {
+  avatarLinkClass: 'my-class',
+  impersonated: 'true',
+  accountName: 'account-name',
+  displayName: 'display-name',
+  logoutPath: 'logout',
+  username: 'username'
+}
+
+describe('<SessionMenu/>', () => {
+  it('renders <SessionMenu/> component when impersonated', () => {
```

Comment:
Specs should describe the functionality of a component. IMO this is not very clear here. Instead of writing a bunch of repetitive and non-flexible assertions, what about writing smaller `it` blocks that describe small pieces of functionality?

For instance:
- If a `.PopNavigation--session` className is required and not just a styling decision, then assert `it should have a X class`
- If there should be more than 1 `.fa-bolt` (which is not clear what this is) then something like `it should have 2 X`
- You could test sign out button in another describe block, with multiple assertions: is it there? what happens when clicked? etc

Reply:
I agree with this. When a component is built up with many pieces, those needs to be described and asserted in separated `it` blocks. It needs to be in this case very descriptive, since apparently there's a lot of pieces that might or not vary depending on props.